.github/workflows/clang-tidy-full.yml

```diff
@@ -0,0 +1,55 @@
+name: Run Linter on all Files
+
+on:
+  workflow_run:
+    workflows: [Build Linter Image]
+    types: [completed]
+    branches: [main]
+  workflow_dispatch:
+
+env:
+  DOCKER_LINTER_DEPENDENCY_IMAGE_NAME: ghcr.io/genspectrum/lapis-silo-linter-dependencies
+
+jobs:
+  waitForDependencyImage:
+    name: Wait for Docker Image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Wait for Docker Image
+        uses: lewagon/wait-on-check-action@v1.3.1
+        with:
+          ref: ${{ github.ref }}
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          check-name: Build linter dependencies
```

Comment:
Do we need this job?
```
on:
  workflow_run:
    workflows: [Build Linter Image]
``` 
should already do the waiting.

Reply:
True. I did not know you can still define branch as a condition. But this does not work for the diffed files unfortunately as I need the PR number in the action to get the list of changed files. And I can only get it in the PR trigger event